### ISSUE TITLE
[FIXED] Bug in the self-attention GNN.

### ISSUE THREAD

**Issue by zgojcic** [overlappredator/OverlapPredator#15]
As correctly noticed by @zhulf0804 (thank you), there was a bug in our implementation of the GNN, which actually led to lower performance of PREDATOR. The bug was fixed in the Pull Request #14. We have now retrained the model on the 3DMatch dataset and obtained a higher performance (see Figure below). We are also retraining PREDATOR on other dataset and will update the tables in the following days. 

Note, due to the change of the GNN architecture old pretrained models will not work anymore, so make sure to always have the latest version.

![image_2021_06_02T09_21_10_068Z](https://user-images.githubusercontent.com/31700292/120461111-06b77b80-c39a-11eb-9f81-c0bbe7486ed6.png)

**Comment by zgojcic** [comment]
Old the 3DMatch models were now retrained after fixing the bug:
The performance of PREDATOR improves up to 8 percent points on the 3DLowMatch:
![3dmatch_result](https://user-images.githubusercontent.com/31700292/121544189-d27f3300-ca09-11eb-9fc1-c976252c3abc.png)

The bug fix also reflects itself in the ablation studies, where the benefit of the OA module and conditioning the features are now even more clear:

![ablation](https://user-images.githubusercontent.com/31700292/121544440-035f6800-ca0a-11eb-8066-80a4fa36af55.png)

![sampling](https://user-images.githubusercontent.com/31700292/121544454-078b8580-ca0a-11eb-89cb-697692520f65.png)

**Comment by zgojcic** [comment]
In the similar manner we are now also retraining the ModelNet and KITTI models and will update the arxiv paper once all results are available.  If you have any questions in the meantime please just contact us directly.

Best,
Zan

**Comment by ShengyuH** [comment]
Updates on Modelnet and KITTI: 

![image](https://user-images.githubusercontent.com/28732721/122232386-7ca00480-cebb-11eb-96aa-a7fdda44e3c7.png)
![image](https://user-images.githubusercontent.com/28732721/122232429-845fa900-cebb-11eb-8fdf-acbc006a7405.png)

arxiv paper and pre-trained weights will be updated in following days.

Best,
Shengyu

**Comment by ShengyuH** [comment]
Updates on MinkowskiEngine-based predator:

<img width="754" alt="image" src="https://user-images.githubusercontent.com/28732721/127786071-be15125b-0e86-48f5-ae1d-50138e7bbfc8.png">

Pretrained weights are updated now, paper will be updated in the following days.

```sh
wget --no-check-certificate --show-progress https://share.phys.ethz.ch/~gsg/Predator/weights.zip
```

best,
Shengyu

**Comment by ShengyuH** [comment]
hi all,

The updated arxiv version of our paper is now available [here](https://arxiv.org/abs/2011.13005). It represents the real performance of our model, please report the updated numbers when comparing to PREDATOR. We will still leave this issue open just in case you miss the updates. 

Best,
the authors

**Comment by zgojcic** [comment]
This issue has now been fixed for all the models with both backbones (MinkowskiEngine and KPConv). The fix increases the performance across all tasks. New values can be seen in the updated version of the [paper](https://arxiv.org/pdf/2011.13005.pdf).

We have also updated all the pretrained models such that it should be possible to reproduce the latest values. 

Thank you for your patience. 

Best,
the authors
